public/app/features/plugins/datasource_srv.ts

```diff
@@ -104,6 +104,7 @@ export class DatasourceSrv implements DataSourceService {
       // The return name or uid needs preservet string containing the variable
       const clone = cloneDeep(dsSettings);
```

Comment:
I know it is not part of this PR, but do we need a deepClone?  Seems like we could return:

```
return {
  ...dsSettings,
  isDefault: false,
  name: key,
  uid: key,
}
```

Reply:
@ryantxu yea think we can, might be a reason for the deepClone but cannot recall one now so let's try without it

Reply:
Could be because it's used somewhere with redux-toolkit that freezes objects?